src/Decompiler/Typing/ExpressionTypeAscender.cs

```diff
@@ -147,8 +147,11 @@ public DataType VisitBinaryExpression(BinaryExpression binExp)
             }
             else
                 throw NYI(binExp);
-            binExp.TypeVariable.DataType = dt;
-            binExp.TypeVariable.OriginalDataType = dt;
+            if (binExp.TypeVariable != null)
+            {
```

Comment:
I have idea to add `storeTypes` parameter to ExpressionTypeAscender constructor. If it is 'false' then data types should not be stored in `TypeVariable`. It will be better than `null` checks